LocalPayment/src/main/java/com/braintreepayments/api/localpayment/LocalPaymentClient.kt

```diff
@@ -0,0 +1,258 @@
+     * @param callback [LocalPaymentAuthCallback]
+     */
+    fun createPaymentAuthRequest(
+        request: LocalPaymentRequest?,
+        callback: LocalPaymentAuthCallback?
+    ) {
+        braintreeClient.sendAnalyticsEvent(LocalPaymentAnalytics.PAYMENT_STARTED)
+
+        var exception: Exception? = null
+
+        if (callback == null) {
+            throw RuntimeException("A LocalPaymentAuthRequestCallback is required.")
+        }
+
+        if (request == null) {
+            exception = BraintreeException("A LocalPaymentRequest is required.")
+        } else if (request.paymentType == null || request.amount == null) {
+            exception = BraintreeException(
+                "LocalPaymentRequest is invalid, paymentType and amount are required."
+            )
+        }
+
+        if (exception != null) {
+            authRequestFailure(exception, callback)
+        } else {
+            braintreeClient.getConfiguration { configuration: Configuration?, error: Exception? ->
+                if (configuration != null) {
+                    if (!configuration.isPayPalEnabled) {
+                        authRequestFailure(
+                            ConfigurationException(
+                                "Local payments are not enabled for this " +
+                                        "merchant."
+                            ),
+                            callback
+                        )
+                        return@getConfiguration
+                    }
+
+                    localPaymentApi.createPaymentMethod(
+                        request!!
+                    ) { localPaymentResult: LocalPaymentAuthRequestParams?, createPaymentMethodError: Exception? ->
+                        if (localPaymentResult != null) {
+                            val pairingId = localPaymentResult.paymentId
+                            if (pairingId != null && !pairingId.isEmpty()) {
+                                payPalContextId = pairingId
+                            }
+                            buildBrowserSwitchOptions(
+                                localPaymentResult,
+                                request.hasUserLocationConsent(),
+                                callback
+                            )
+                        } else if (createPaymentMethodError != null) {
+                            authRequestFailure(
+                                BraintreeException(
+                                    "An error occurred creating the " +
+                                            "local payment method."
+                                ),
+                                callback
+                            )
+                        }
+                    }
+                } else if (error != null) {
+                    authRequestFailure(error, callback)
+                }
+            }
+        }
+    }
+
+    fun buildBrowserSwitchOptions(
```

Comment:
Does this function need to be public? It seems to be package-private in the java file. I'm not sure where this is being called, but if it's only called in the same module, we could mark it as `internal`. If other modules are invoking it, we can add the `@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)` annotation to prevent merchants from calling it.